lib/AST/NameLookup.cpp

```diff
@@ -2434,6 +2434,49 @@ CustomAttrNominalRequest::evaluate(Evaluator &evaluator,
     }
   }
 
+  // If we have more than one attribute declaration, we have an ambiguity.
+  // So, emit an ambiguity diagnostic.
+  if (nominals.size() > 1) {
+    SmallVector<NominalTypeDecl *, 4> ambiguousCandidates;
+    // Filter out declarations that cannot be property wrappers.
+    for (auto decl : nominals) {
+      if (isa<ProtocolDecl>(decl)) {
+        continue;
+      }
+      ambiguousCandidates.push_back(decl);
+    }
+    if (ambiguousCandidates.size() > 1) {
+      auto attrName = nominals.front()->getName();
+      ctx.Diags.diagnose(attr->getLocation(),
+                         diag::ambiguous_custom_attribute_ref, attrName);
+      for (auto candidate : ambiguousCandidates) {
+        ctx.Diags.diagnose(candidate->getLoc(),
+                           diag::found_attribute_candidate);
+        // If the candidate is a top-level property wrapper, let's suggest
+        // adding module name to resolve the ambiguity.
+        if (candidate->getDeclContext()->isModuleScopeContext()) {
+          auto moduleName = candidate->getParentModule()->getName();
+          ctx.Diags.diagnose(attr->getLocation(),
+                             diag::ambiguous_custom_attribute_ref_fix,
+                             moduleName.str(), attrName, moduleName);
```

Comment:
Awesome! Can we add a Fix-It too?

Reply:
Oops, I added it but forgot to commit it!

Reply:
Done ✅